packages/react-components/src/collection/src/useAsyncSearch.ts

```diff
@@ -34,9 +34,9 @@ export function useAsyncSearch(load) {
                 setItems(result ?? []);
                 setIsLoading(false);
             })
-            .catch(error => {
+            .catch((error: unknown) => {
                 // To cancel a promise it must be rejected, ignore it. If it's something else, show no results.
-                if (isNil(error) || error?.isCancelled !== true) {
+                if (isNil(error) || (error as CancellablePromiseError)?.isCancelled !== true) {
```

Comment:
I've change it to use the proper way. Catch clause should always be typed as any (will be enforce soon in the next version of TS). Since we can throw errors and/or any object litterals, we can't know the type of the error. We should check the type of the error before using it